src/vs/platform/files/common/files.ts

```diff
@@ -515,20 +542,28 @@ export class FileChangesEvent {
 			return false;
 		}
 
-		const hasTypesFilter = types.length > 0;
+		// Added
+		if (!types || types.includes(FileChangeType.ADDED)) {
```

Comment:
`types` is never undefined, instead it can be empty. So you need to check for `!types.length` instead of `!types`. This will fix failing unit tests.

Reply:
@sandy081 thanks for catching this!